### ISSUE TITLE
NoMethodError in Chrome (undefined method `strip' for nil:NilClass)

### ISSUE THREAD

**Issue by clockworkpc** [titusfortner/webdrivers#67]
### System
Ubuntu 18.04
Ruby 2.5.1p57 (freshly installed)
NOTE: I also tried this on Ruby 2.6.2 and 2.5.5
Chromium version: Version 73.0.3683.75 (Official Build) Built on Ubuntu , running on Ubuntu 18.04 (64-bit)

### Steps Followed
Created a brand new rails app with default settings.
Here is my Gemfile:
[Gemfile.txt](https://github.com/titusfortner/webdrivers/files/3078943/Gemfile.txt)

### Chrome Browser (Fails)
```
Running via Spring preloader in process 29442
Loading development environment (Rails 5.2.3)
irb(main):001:0> Watir::Browser.new(:chrome)
rails_console: No such file or directory - --product-version
Traceback (most recent call last):
        2: from (irb):1
        1: from (irb):1:in `new'
NoMethodError (undefined method `strip' for nil:NilClass)
```

### Firefox Browser (Works OK)
```
irb(main):003:0> b = Watir::Browser.new(:firefox)
=> #<Watir::Browser:0x3a74db97bfeb780e url="about:blank" title="">
irb(main):005:0> b.goto('www.google.com')
=> "http://www.google.com"
irb(main):006:0>
```

### Current Solution: Go back to `chromedriver-helper` for Chrome
If I include `gem 'chromedriver-helper'` in my Gemfile, run `bundle install` and jump back into the Rails console, then Chrome works too.

```
Running via Spring preloader in process 29466
Loading development environment (Rails 5.2.3)
irb(main):001:0> b = Watir::Browser.new(:chrome)
=> #<Watir::Browser:0x..f9dd5c620c38cd000 url="data:," title="">
irb(main):002:0> b.goto('www.google.com')
=> "http://www.google.com"
```

Firefox is fine either way.

**Comment by kapoorlakshya** [comment]
Hi @clockworkpc, please try providing the path to the Chromium binary as described [here](https://github.com/titusfortner/webdrivers#when-using-chromechromium).

**Comment by clockworkpc** [comment]
> Hi @clockworkpc, please try providing the path to the Chromium binary as described [here](https://github.com/titusfortner/webdrivers#when-using-chromechromium).

Thanks for the reply.  I checked my Chromium path and found that it indeed differs from the standard installation path:

```
$ which chromium-browser 
/usr/bin/chromium-browser
```

This was never a problem with `chromedriver-helper`, but now that I know it's no big deal.

I created a small file: `config/chromium_path.rb` which has the following line:
`Selenium::WebDriver::Chrome.path = '/usr/bin/chromium-browser'`

This seems to have done the trick.  Is there a better way to do this?

**Comment by kapoorlakshya** [comment]
> This was never a problem with chromedriver-helper

Actually, `webdrivers` didn't need the browser binary path either until we added support for `chromedriver` 70+. Google started matching the browser and driver versions which now requires us to know the browser version before we can download the correct driver. See http://chromedriver.chromium.org/downloads/version-selection.

 `chromedriver-helper` would have required a similar logic for versions 70+. 

> Is there a better way to do this?

Hopefully soon :). This is the only way until we implement #45.

I'm closing the issue, but feel free to continue the discussion if needed.

**Comment by andretf** [comment]
TLDR;
`# ln -s /usr/bin/google-chrome-stable /usr/bin/google-chrome`

I had the same error been throwing from here:
https://github.com/titusfortner/webdrivers/blob/bee776a18b23d8fba0478c86fcea1056eff4fd23/lib/webdrivers/chromedriver.rb#L135

That is, I'm using Google Chrome on Linux, more specifically, a Manjaro with Chrome from [AUR](https://aur.archlinux.org/packages/google-chrome/). Therefore, what I have in my PATH is a script named `google-chrome-stable` rather than `google-chrome` binary or symlink.

Looking at #45, and further https://cs.chromium.org/chromium/src/chrome/test/chromedriver/chrome/chrome_finder.cc, I don't think this issue would be solved for me, and a bunch of people.

If this case won't be handled, worth it put such a reminder in #README.

**Comment by kapoorlakshya** [comment]
@andretf Does it work if you set `Selenium::WebDriver::Chrome.path = '/usr/bin/google-chrome'`?

**Comment by durrantm** [comment]
I am experiencing this using Docker on a Ubuntu 19.04 image (desktop is 18) I do not have `/usr/bin/google-chrome` in my tree.  `chromedriver` is in `/usr/bin/chromedriver`  I also keep getting `DevToolsActivePort file doesn't exist` messages

**Comment by kapoorlakshya** [comment]
@durrantm Point `Selenium::WebDriver::Chrome.path` to wherever Chrome/Chromium is installed and remove any other chromedriver binary from your `PATH`.

I'll put in a PR to improve this error message.

**Comment by andretf** [comment]
@kapoorlakshya 
It works if I set `Selenium::WebDriver::Chrome.path = '/usr/bin/google-chrome-stable'`, is that what you did mean? Pointing to `/usr/bin/google-chrome` would throw an error.

**Comment by kapoorlakshya** [comment]
@andretf Use the binary that responds to `--product-version` by returning the Chrome version. If `/usr/bin/google-chrome-stable --product-version` returns the version instead of launching the browser or throwing an error, I would use it.

**Comment by ngollan** [comment]
If you want to use Chromium on Linux (which obviously nobody ever tested either before releasing 3.9.3 to realise that the "fancy" `System.execute`—that would probably break if whitespace happened, please look up `Open3`—can return `nil`):

    update-alternatives --install /usr/bin/google-chrome chrome-browser $(which chromium-browser) 1
